diff --git a/lib/timer/rte_timer.c b/lib/timer/rte_timer.c
index bb8b6a651..44298a886 100644
--- a/lib/timer/rte_timer.c
+++ b/lib/timer/rte_timer.c
@@ -24,7 +24,7 @@
 /**
  * Per-lcore info for timers.
  */
-struct __rte_cache_aligned priv_timer {
+struct priv_timer {
 	struct rte_timer pending_head;  /**< dummy timer instance to head up list */
 	rte_spinlock_t list_lock;       /**< lock to protect list access */
 
@@ -44,7 +44,7 @@ struct __rte_cache_aligned priv_timer {
 	/** per-lcore statistics */
 	struct rte_timer_debug_stats stats;
 #endif
-};
+} __rte_cache_aligned;
 
 #define FL_ALLOCATED	(1 << 0)
 struct rte_timer_data {
@@ -213,6 +213,20 @@ rte_timer_init(struct rte_timer *tim)
 	rte_atomic_store_explicit(&tim->status.u32, status.u32, rte_memory_order_relaxed);
 }
 
+int
+rte_timer_meta_init(void)
+{
+    struct rte_timer_data *timer_data;
+    struct priv_timer *pt;
+    unsigned lcore_id = rte_lcore_id();
+
+    TIMER_DATA_VALID_GET_OR_ERR_RET(default_data_id, timer_data, -EINVAL);
+    pt = &timer_data->priv_timer[lcore_id];
+    memset(pt, 0, sizeof(*pt));
+    pt->prev_lcore = lcore_id;
+    return 0;
+}
+
 /*
  * if timer is pending or stopped (or running on the same core than
  * us), mark timer as configuring, and on success return the previous
diff --git a/lib/timer/rte_timer.h b/lib/timer/rte_timer.h
index 3c7d236c2..882bc4660 100644
--- a/lib/timer/rte_timer.h
+++ b/lib/timer/rte_timer.h
@@ -190,6 +190,15 @@ void rte_timer_subsystem_finalize(void);
  */
 void rte_timer_init(struct rte_timer *tim);
 
+/**
+ * Initialize the timer metadata.
+ * For f-stack internal use only.
+ * @return
+ *   - 0: Success
+ *   - -EINVAL: invalid timer data instance identifier
+ */
+int rte_timer_meta_init(void);
+
 /**
  * Reset and start the timer associated with the timer handle.
  *
